playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

```diff
@@ -312,6 +329,8 @@ private void preparePlayer(@NonNull PKMediaSourceConfig sourceConfig) {
         metadataList.clear();
         isLoadedMetaDataFired = false;
         shouldGetTracksInfo = true;
+        // Need to clear the overrides of DefaultTrackSelector before loading the next media.
+        trackSelectionHelper.clearPreviousMediaOverrides();
```

Comment:
please just verify when we change media and we have textTrack preffered language it will still work

Reply:
Preferred track logic is working because that selection is done once we build the tracks.